### PR TITLE
Component refactor

### PR DESCRIPTION
**Pull request by koredefashokun**
## Description
This PR refactors components to optimize performance and code readability. It also moves some state to Redux as well.

## Related issue
N/A

## Motivation and Context
N/A

## How Has This Been Tested?
N/A

### CODE REVIEW

File: sections/exchange/MobileSwap/RatioSelect.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ const RatioSelect: FC = () => {
 				<RatioButton
 					key={`ratio-${v}`}
 					$selected={ratio === v}
-					onClick={() => onRatioChange(v)}
+					onClick={onRatioChange(v)}
```
Comment (avclarke) [inline sections/exchange/MobileSwap/RatioSelect.tsx:29]: Not sure I agree with this change as it's a pattern not used anywhere else throughout the codebase so presumed it was a bug until I seen you added the call step above. Would prefer if we reverted to remain consistent.

File: state/earn/selectors.ts [hunk 2]
```diff
@@ -1,12 +1,35 @@
 import { createSelector } from '@reduxjs/toolkit';
+import { wei } from '@synthetixio/wei';
 
 import { RootState } from 'state/store';
-import { toWei } from 'utils/formatters/number';
+import { truncateNumbers } from 'utils/formatters/number';
+
+const SECONDS_PER_DAY = 86400;
```
Comment (avclarke) [inline state/earn/selectors.ts:7]: Seems we already have this constant declared in a few places if you wanted to import instead

**Review by avclarke — COMMENTED** [review]
Nice changes! Just a couple of minor suggestions

File: sdk/services/exchange.ts [hunk 3]
```diff
@@ -734,46 +723,42 @@ export default class ExchangeService {
...
 		const balanceCalls = [];
 
 		for (const addr of deprecatedProxySynthsAddresses) {
 			symbolCalls.push(getProxySynthSymbol(addr));
-			balanceCalls.push(Redeemer.balanceOf(addr, this.sdk.context.walletAddress));
+			balanceCalls.push(Redeemer.balanceOf(addr, walletAddress));
 		}
 
 		const [deprecatedSynths, balanceData] = (await Promise.all([
 			this.sdk.context.multicallProvider.all(symbolCalls),
```
Comment (avclarke) [inline sdk/services/exchange.ts:749]: We could combine and then split these to have them in one call

File: state/earn/selectors.ts (hunk 2 above)
Comment (koredefashokun) [reply state/earn/selectors.ts:7]: Actually, we don't seem to have it declared anywhere.

Comment (avclarke) [reply state/earn/selectors.ts:7]: There's one in `sdk/constants/period`

File: sdk/services/exchange.ts (hunk 3 above)
Comment (koredefashokun) [reply sdk/services/exchange.ts]: Done

File: state/earn/selectors.ts (hunk 2 above)
Comment (koredefashokun) [reply state/earn/selectors.ts:7]: My bad. Was searching for `86400` instead.

File: sections/exchange/MobileSwap/RatioSelect.tsx (hunk 1 above)
Comment (koredefashokun) [reply sections/exchange/MobileSwap/RatioSelect.tsx]: Changed.

File: components/StakeCard/StakeCard.tsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+	unstakeBalance: Wei;
+	onStake(amount: string): void;
+	onUnstake(amount: string): void;
+	stakeEnabled?: boolean;
+	unstakeEnabled?: boolean;
+	isApproved?: boolean;
+	onApprove?: () => void;
+};
+
+const StakeCard: FC<StakeCardProps> = ({
```
[the remark this answers is not in the document]
Comment (koredefashokun) [reply components/StakeCard/StakeCard.tsx:26]: Done.

File: sdk/services/exchange.ts [hunk 5]
```diff
@@ -734,46 +723,42 @@ export default class ExchangeService {
...
 		])) as [CurrencyKey[], ethers.BigNumber[]];
 
-		const balances = balanceData.map((balance) => wei(balance));
-
 		let totalUSDBalance = wei(0);
 		const cryptoBalances: DeprecatedSynthBalance[] = [];
 
-		for (let i = 0; i < balances.length; i++) {
-			const usdBalance = balances[i];
+		for (let i = 0; i < balanceData.length; i++) {
```
[the remark this answers is not in the document]
Comment (avclarke) [reply sdk/services/exchange.ts]: Oh yea nice!

**Review by avclarke — APPROVED** [review]

**Review by platschi — APPROVED** [review]
Blazing fast. Great work!

**Pull request merged**